discovery/discovery_test.go

```diff
@@ -1193,18 +1195,88 @@ func TestOrchestratorPool_GetOrchestrators(t *testing.T) {
 
 }
 
+func TestOrchestratorPool_GetOrchestrators_SuspendedOrchs(t *testing.T) {
+	assert := assert.New(t)
+	require := require.New(t)
+	addresses := stringsToURIs([]string{"https://127.0.0.1:8936", "https://127.0.0.1:8937", "https://127.0.0.1:8938"})
+
+	wg := sync.WaitGroup{}
+
+	orchCb := func() error { return nil }
+	oldOrchInfo := serverGetOrchInfo
+	defer func() { wg.Wait(); serverGetOrchInfo = oldOrchInfo }()
+	serverGetOrchInfo = func(ctx context.Context, bcast common.Broadcaster, server *url.URL) (*net.OrchestratorInfo, error) {
+		defer wg.Done()
+		err := orchCb()
+		return &net.OrchestratorInfo{
+			Transcoder: server.String(),
+		}, err
+	}
+
+	pool := NewOrchestratorPool(nil, addresses)
+
+	// suspend https://127.0.0.1:8938
+	sus := newStubSuspender()
+	sus.list["https://127.0.0.1:8938"] = 5
+	require.Greater(sus.Suspended("https://127.0.0.1:8938"), 0)
+
+	// don't include suspended orchestrators if enough orchestrators are available
+	wg.Add(len(addresses))
+	res, err := pool.GetOrchestrators(2, sus)
+	assert.Nil(err)
+	assert.Len(res, 2)
+	assert.NotEqual(res[0].GetTranscoder(), "https://127.0.0.1:8938")
+	assert.NotEqual(res[1].GetTranscoder(), "https://127.0.0.1:8938")
+
+	// include suspended O's if not enough non-suspended O's available
+	wg.Add(len(addresses))
+	require.Greater(sus.Suspended("https://127.0.0.1:8938"), 0)
+	res, err = pool.GetOrchestrators(3, sus)
+	assert.Nil(err)
+	assert.Len(res, 3)
+	// suspended Os are added last
+	assert.Equal(res[2].Transcoder, "https://127.0.0.1:8938")
+
+	// no suspended O's, insufficient non-suspended O's
```

Comment:
Doesn't seem like this comment is accurate since there are still suspended Os in this test case since the same instance of `sus` is used and we suspended `https://127.0.0.1:8938` on L1220?